### PR TITLE
[no ticket][risk=no] e2e code improvement

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: e2e/app/component/data-resource-card.ts [hunk 1]
```diff
@@ -47,23 +47,20 @@ export default class DataResourceCard extends CardBase {
...
+    return page
+      .waitForXPath(selector, { timeout })
+      .then((element) => {
+        logger.info(`Found data resource card: "${resourceName}"`);
+        return new DataResourceCard(page).asCard(element);
+      })
+      .catch(() => {
+        logger.info(`Data resource card: "${resourceName}" is not found`);
+        return null;
+      });
```
Comment (aweng98) [inline e2e/app/component/data-resource-card.ts:63]: Simplify func: avoid use of `findAllCards` func then iterate over a loop.

File: e2e/app/component/workspace-card.ts [hunk 2]
```diff
@@ -74,20 +74,20 @@ export default class WorkspaceCard extends CardBase {
...
+    return page
+      .waitForXPath(selector, { timeout })
+      .then((element) => {
+        logger.info(`Found workspace card: "${workspaceName}"`);
+        return new WorkspaceCard(page).asCard(element);
+      })
+      .catch(() => {
+        logger.info(`Workspace card: "${workspaceName}" is not found`);
+        return null;
+      });
```
Comment (aweng98) [inline e2e/app/component/workspace-card.ts:90]: Same as before: Simplified func.

File: e2e/app/modal/share-modal.ts [hunk 3]
```diff
@@ -48,9 +48,9 @@ export default class ShareModal extends Modal {
       const ownerOpt = await this.waitForRoleOption(level);
       await ownerOpt.click();
     };
-    const typeAndAddUser = async (): Promise<boolean> => {
+    const typeAndAddUser = async (name): Promise<boolean> => {
```
Comment (aweng98) [inline e2e/app/modal/share-modal.ts:51]: Pass `name` string to the `typeAndAddUser`.

File: e2e/app/page/workspace-base.ts [hunk 4]
```diff
@@ -98,7 +98,7 @@ export default abstract class WorkspaceBase extends AuthenticatedPage {
     const selector = buildXPath({ name: pageTabName, type: ElementType.Tab });
     const tabLink = new Link(this.page, selector);
     if (!(await tabLink.exists())) {
-      throw new Error(`Failed to find and click \"${pageTabName}\" page tab.`);
+      throw new Error(`Failed to find and click "${pageTabName}" page tab.`);
```
Comment (aweng98) [inline e2e/app/page/workspace-base.ts:101]: Unnecessary `"`.

File: e2e/tests/datasets/export-to-notebook.spec.ts [hunk 5]
```diff
@@ -80,8 +80,7 @@ describe('Export dataset to notebook tests', () => {
    * Test:
    * - Create dataset and export to notebook. Start the notebook and run the dataset code.
    */
-  // TODO Enable notebook test after bug fix. https://precisionmedicineinitiative.atlassian.net/browse/RW-6885
-  xtest.each(KernelLanguages)('Export to %s kernel Jupyter notebook when creating dataset', async (kernelLanguage) => {
+  test.each(KernelLanguages)('Export to %s kernel Jupyter notebook when creating dataset', async (kernelLanguage) => {
```
Comment (aweng98) [inline e2e/tests/datasets/export-to-notebook.spec.ts:83]: Bug was closed.

File: e2e/tests/nightly/dataproc-to-gce.spec.ts [hunk 6]
```diff
@@ -6,9 +6,9 @@ import { makeRandomName } from 'utils/str-utils';
 import { createWorkspace, signInWithAccessToken } from 'utils/test-utils';
 
 // This test could take a long time to run
-jest.setTimeout(60 * 30 * 1000);
+jest.setTimeout(40 * 60 * 1000);
 // Retry one more when fails
-jest.retryTimes(0);
+jest.retryTimes(1);
```
Comment (aweng98) [inline e2e/tests/nightly/dataproc-to-gce.spec.ts:11]: adds back retry if fail and longer test timeout from 30 min to 40 min.

File: e2e/app/component/data-resource-card.ts [hunk 7]
```diff
@@ -47,23 +47,20 @@ export default class DataResourceCard extends CardBase {
...
-    }
-    return null; // not found
+  static async findCard(page: Page, resourceName: string, timeout = 30000): Promise<DataResourceCard | null> {
+    const selector =
+      `${DataResourceCardSelector.cardRootXpath}[.//*[${DataResourceCardSelector.cardNameXpath}` +
+      ` and normalize-space(text())="${resourceName}"]]`;
+    return page
+      .waitForXPath(selector, { timeout })
+      .then((element) => {
+        logger.info(`Found data resource card: "${resourceName}"`);
```
Comment (petesantos) [inline e2e/app/component/data-resource-card.ts:57]: How often do we use the log information for the these scenarios ?

**Review by petesantos — APPROVED** [review]
looks good

File: e2e/app/component/data-resource-card.ts (hunk 7 above)
Comment (aweng98) [reply e2e/app/component/data-resource-card.ts:57]: I use log for test failure investigation often. It's verbose but useful. Thank you very much.

**Pull request merged**